core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -52,6 +52,14 @@ public static TableMetadata newTableMetadata(Schema schema,
                                                PartitionSpec spec,
                                                String location,
                                                Map<String, String> properties) {
+    return newTableMetadata(schema, spec, location, properties, DEFAULT_TABLE_FORMAT_VERSION);
+  }
+
+  static TableMetadata newTableMetadata(Schema schema,
+                                        PartitionSpec spec,
+                                        String location,
+                                        Map<String, String> properties,
+                                        int formatVersion) {
```

Comment:
This is needed to create v2 tables in tests. It isn't public.

Reply:
Is it something we want to annotate with @VisibleForTesting?

Reply:
We don't use that annotation elsewhere and I'm reluctant to start because it increases our dependence on guava. It is also likely that we will open this up later, so I don't think we need to.